### PR TITLE
#6095 rename factory functions to match new types

### LINKED ISSUE #6095: Rename factory functions to match new types
## Scope
- Rename all factory functions to match the associated type
  - Ex: `src/testUtils/factories/extensionFactories.ts`

## Additional
- Rename BrickIcon -> PackageIcon

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Closes #6095

## Checklist

- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/testUtils/factories/recipeFactories.ts
Comment (twschiller) [inline src/testUtils/factories/recipeFactories.ts:1]: This file should be renamed to modDefinitionFactories

File: src/pageEditor/starterBricks/base.test.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ import {
 import { type StarterBrickType } from "@/starterBricks/types";
 import { type ReaderConfig } from "@/bricks/types";
 import { validateRegistryId } from "@/types/helpers";
-import { extensionPointDefinitionFactory } from "@/testUtils/factories/recipeFactories";
+import { starterBrickConfigFactory as starterBrick } from "@/testUtils/factories/recipeFactories";
```
Comment (twschiller) [inline src/pageEditor/starterBricks/base.test.ts:26]: This is an odd rename

Comment (grahamlangford) [reply src/pageEditor/starterBricks/base.test.ts:26]: I have no idea where that came from. Fixing.

**Comment by grahamlangford** [comment]
> 

done

**Comment by grahamlangford** [comment]
> @fregante looking good overall. I think we need to rename recipeFactory -> modDefinitionFactory file

done

**Pull request merged**